lib/response.js

```diff
@@ -455,17 +455,22 @@ internals.Response.prototype._prepare = function (next) {
 
     const onDone = (source) => {
 
-        if (source instanceof Error) {
-            return next(Boom.wrap(source));
-        }
+        // do not use Hoek.nextTick because it will not work as intended
```

Comment:
See https://github.com/hapijs/hapi/pull/3202/files#diff-d86a59ede7d999db4b7bc43cb25a1c11R399, it doesn't have to be weird :)